src/scripts/create_elner.py

```diff
@@ -0,0 +1,222 @@
+# /// script
+# requires-python = ">=3.10,<4.0"
+# dependencies = [
+#     "datasets==3.5.0",
+#     "huggingface-hub==0.24.0",
+#     "pandas==2.2.0",
+#     "requests==2.32.3",
+# ]
+# ///
+
+"""Create the elNER dataset and upload it to the HF Hub."""
+
+import io
+import tarfile
+from collections import defaultdict
+
+import pandas as pd
+import requests
+from datasets import Split
+from datasets.arrow_dataset import Dataset
+from datasets.dataset_dict import DatasetDict
+from huggingface_hub.hf_api import HfApi
+
+
+def main() -> None:
+    """Create the elNER dataset and uploads it to the HF Hub."""
+    # Define the download URL for the tar.gz file
+    url = "https://github.com/nmpartzio/elNER/raw/master/dataset/elNER18/IOB2.tar.gz"
+
+    # Download the dataset
+    response = requests.get(url)
+    response.raise_for_status()
+
+    # Extract the tar.gz file
+    tar_file = tarfile.open(fileobj=io.BytesIO(response.content), mode="r:gz")
+
+    data = read_files_from_tar(tar_file=tar_file)
+
+    dfs = parse_iob2(data=data)
+
+    # Create splits
+    train_size = 1024
+    val_size = 256
+    test_size = 2048
+
+    val_df = dfs["val"].sample(n=val_size, random_state=4242)
+    test_df = dfs["test"].sample(n=test_size, random_state=4242)
+    train_df = dfs["train"].sample(n=train_size, random_state=4242)
+
+    # Rename ner_tags to labels for consistency
+    train_df = train_df.rename(columns={"ner_tags": "labels"}).reset_index(drop=True)
+    val_df = val_df.rename(columns={"ner_tags": "labels"}).reset_index(drop=True)
+    test_df = test_df.rename(columns={"ner_tags": "labels"}).reset_index(drop=True)
+
+    # Convert the NER tags to standard format
+    ner_conversion_dict = {
+        # Keep O as is
+        "O": "O",
+        # Person entities
+        "B-PERSON": "B-PER",
+        "I-PERSON": "I-PER",
+        # Organization entities
+        "B-ORG": "B-ORG",
+        "I-ORG": "I-ORG",
+        # Location entities (including GPE and FAC)
+        "B-LOC": "B-LOC",
+        "I-LOC": "I-LOC",
+        "B-GPE": "B-LOC",  # Geopolitical entity
+        "I-GPE": "I-LOC",
+        "B-FAC": "B-LOC",  # Facility
+        "I-FAC": "I-LOC",
+        # Miscellaneous entities
+        "B-NORP": "B-MISC",  # Nationality, religion, political group
+        "I-NORP": "I-MISC",
+        "B-DATE": "B-MISC",
+        "I-DATE": "I-MISC",
+        "B-TIME": "B-MISC",
+        "I-TIME": "I-MISC",
+        "B-QUANTITY": "B-MISC",
+        "I-QUANTITY": "I-MISC",
+        "B-CARDINAL": "B-MISC",
+        "I-CARDINAL": "I-MISC",
+        "B-ORDINAL": "B-MISC",
+        "B-PERCENT": "B-MISC",
+        "B-MONEY": "B-MISC",
+        "I-MONEY": "I-MISC",
+        "B-EVENT": "B-MISC",
+        "I-EVENT": "I-MISC",
+        "B-WORK_OF_ART": "B-MISC",
+        "I-WORK_OF_ART": "I-MISC",
+        "B-PRODUCT": "B-MISC",
+        "I-PRODUCT": "I-MISC",
+        "B-LAW": "B-MISC",
+        "I-LAW": "I-MISC",
+        "B-LANGUAGE": "B-MISC",
```

Comment:
Some of these are so common that I'd just replace them with "O" - we've done that with the [DANSK dataset as well](https://euroeval.com/datasets/danish/#dansk), for instance.

```suggestion
        # Miscellaneous entities
        "B-NORP": "B-MISC",  # Nationality, religion, political group
        "I-NORP": "I-MISC",
        "B-DATE": "O",
        "I-DATE": "O",
        "B-TIME": "O",
        "I-TIME": "O",
        "B-QUANTITY": "O",
        "I-QUANTITY": "O",
        "B-CARDINAL": "O",
        "I-CARDINAL": "O",
        "B-ORDINAL": "O",
        "B-PERCENT": "O",
        "B-MONEY": "O",
        "I-MONEY": "O",
        "B-EVENT": "B-MISC",
        "I-EVENT": "I-MISC",
        "B-WORK_OF_ART": "B-MISC",
        "I-WORK_OF_ART": "I-MISC",
        "B-PRODUCT": "B-MISC",
        "I-PRODUCT": "I-MISC",
        "B-LAW": "O",
        "I-LAW": "O",
        "B-LANGUAGE": "B-MISC",
```